api/proto/teleport/autoupdate/v1/autoupdate.proto

```diff
@@ -83,6 +83,10 @@ message AgentAutoUpdateGroup {
   // wait_hours after last group succeeds before this group can run. This can only be used when the strategy is "halt-on-failure".
   // This field must be positive.
   int32 wait_hours = 5;
+  // canary_count is the number of canary agents that will be updated before the whole group is updated.
+  // when set to 0, the group does not enter the canary phase. This number is capped to 5.
+  // This number must always be lower than the total number of agents in the group, else the rollout will be stuck.
+  int64 canary_count = 6;
```

Comment:
Nit: Just curious if there's specific reason this is `int64`? Seems like this will always be a small-ish number.